### DISCUSSION TITLE
Single atomic method for two set() and expireAt() calls?

### DISCUSSION THREAD

**Discussion by babakeghbali** [xetorthio/jedis#2606] [Q&A] [answered]
Hi,

I know the atomic **setex()** method can be used to set a key with a TTL value like this:

    `jedis.setex(key, 3600,value);`

But I could not find the equivalent method for setting a expiration timestamp instead of TTL value. I currently have following actions, but looking for an atomic method like **setex()**:

```
jedis.set(key, value);
jedis.expireAt(key,timestamp); 
```

Does it exist and I'm missing it or?

Thanks in advance.

**Comment by sazzad16** [comment, marked as answer]
Use SET command with EXAT argument.
```
jedis.set(key, value, SetParams.setParams().exAt(timestamp));
```

**Reply by babakeghbali** [reply]
Thank you @sazzad16 for the answer. 
The set() method with params can be used in the same way to set TTL, but setex() was introduced too.
Now just out of curiosity, do you have any idea why such thing does not exist for expiration timestamp?

**Reply by sazzad16** [reply]
@babakeghbali If you see the documentation of SET https://redis.io/commands/set command, there are many arguments. To cover all the valid variations among the arguments, about 24* different commands would be required. Now imagine adding newer arguments; this number would be multiplied and increase exponentially. This is convenient for neither the developers not the users. So Redis opted for single command and optional arguments.

(\* This is my quick calculation. May not be exactly correct.)

---

SETEX is quite old command. This and other old commands are kept for backward compatibility.

**Reply by babakeghbali** [reply]
True @sazzad16  , makes sense. Thanks.
